src/webgpu/util/device_pool.ts

```diff
@@ -69,7 +69,10 @@ export class DevicePool {
           holder.device.destroy();
         }
       }
-      throw ex;
+      // Supress failures if the device was explicitly destroyed.
+      if (holder.lostInfo !== undefined && holder.lostInfo.reason !== 'destroyed') {
+        throw ex;
+      }
```

Comment:
which failures happen? isn't it caught in expectValidationError ?

Reply:
If device is lost, there is a separate check, not just the validation check. Before the code re-threw the error. Now we only re-throw the error if the device wasn't explicitly destroyed.

Reply:
Ah I see - however, my concern with this is that it will not throw for *any* device.destroy() call in *any test*.
I think it would be better if we can explicitly tell the harness in the test that we're expecting the lostInfo.reason to be "destroyed"

Reply:
Changed it so that it requires an explicit call.